test/units/parsing/test_ajson.py

```diff
@@ -20,3 +24,147 @@ def test_AnsibleJSONDecoder_vault():
     assert isinstance(data['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['bar']['baz'][0]['password'], AnsibleVaultEncryptedUnicode)
     assert isinstance(data['foo']['password'], AnsibleVaultEncryptedUnicode)
+
+
+def vault_data():
```

Comment:
This function must be a fixture. You invoke it multiple times in tests which causes rereading the same file. Which in turn slows down tests execution.
```suggestion
@pytest.fixture
def vault_data():
```

Reply:
I tried to pass it as a fixture but failed. If you know a working way to pass it in the decorator as a fixture, please, share it

Reply:
In the decorator? What do you mean?

Reply:
seems we fixed it by another way